coderd/autobuild/executor/lifecycle_executor.go

```diff
@@ -34,22 +42,42 @@ func New(ctx context.Context, db database.Store, log slog.Logger, tick <-chan ti
 	return le
 }
 
+// WithStatsChannel will cause Executor to push a RunStats to ch after
+// every tick.
+func (e *Executor) WithStatsChannel(ch chan<- RunStats) *Executor {
+	e.statsCh = ch
+	return e
+}
+
```

Comment:
Should we always do this?

Reply:
I don't have any major use-case for consuming these outside of the unit tests, hence the toggle.
Thoughts?

Reply:
Fair enough. I think it's fine for now!